lnst/Devices/Device.py

```diff
@@ -532,6 +533,13 @@ def ips(self):
         """
         return self._ip_addrs
 
+    @property
+    def lnst_ips(self) -> list[BaseIpAddress]:
```

Comment:
i get what the goal is here, but i'm not 100% sure about placing this property into the `Device` class.

My main reason is that the "abstract idea" behind the Device class is to provide a "get and push" interface for the device configuration on the system. These two operations being separated but always representing the "current/immediate" state - a push operation immediately sends the configuration request, and a get operation retrieves the current most up to date information.

This `lnst_ips` property however is... static and represents an "expected" state during some time of the test recipe. There can be situations where this is not the TRUE state of the system. An example of such a situation being if there's some configuration action that happens outside of the scope of the `Device` class. For example if a recipe calls some kind of a `/sys/` path configuration that for some reason initiates a reset of the device configuration. In this case this property stays static, and that's GOOD - when we generate endpoints I want that to be static based on what we THINK we configured. However, if this property is placed in the `Device` class I think it may be confusing as everything else in the class represents the immediate state.

And at the same time I think that this implementation of the `lnst_ips` may not be fully sufficient for all of our test recipes. For example, some recipes step outside of the `Device` classes for their configuration, such as the DPDK recipes, because in these cases there aren't really any "configured" ip addresses since the device driver module is unloaded, however there are "imagined" configured ip addresses that should be used for the Endpoint generation.

Reply:
Right, could we maybe change all recipes to manually store each configured IP in `recipe_conf`, or make `Device` wrapped in something, BUT AFAICS no DPDK recipe inherits from `BaseFlowMeasurementGenerator` and thus doesn't define `generate_perf_endpoints`. The `OvSDPDKBondRecipe` *does* define this method, but I think that's a mistake, because it's probably not called anywhere - the return value also doesn't make any sense, because it returns `DummyRecipeConfig`s, which don't define `ips_filter`, which is used to filter the IPs from endpoints currently.

As for `Device` being an immediate state... I'm also against mixing it, but at the same time I would expect each recipe to have access to its configured NICs and each NIC have access to its configured IPs. Maybe we should rename the property to something like `configured_lnst_ips` so its clear we expect them to be *configured* as opposed to actually being there I guess. I'm not sure how else to tackle this. I wouldn't really want to create a wrapper or a different structure for just IPs.

Reply:
Not relevant anymore.